### ISSUE TITLE
Select overflow

### ISSUE THREAD

**Issue by ldaug** [PBGUX/bower-designsystem#8]
I have a select box in which I am feeding in dynamic names at run time, some of which can be quite long (labels from data hub items). The select is in a confined space so the default just chops off the labels if they are too long; there is no scrollbar on the x axis. Setting overflow to auto fixes this, but unfortunately when an item is selected it again gets truncated about 10 or 15px from the right hand side.

![image](https://cloud.githubusercontent.com/assets/7534439/19171864/f568edd0-8bec-11e6-89b2-9d9afcc062f1.png)

**Comment by danielcollettepb** [comment]
@ldaug can you send a sample of the data... I want to mock this up in a plnkr to see what can be done. Thanks...

**Comment by danielcollettepb** [comment]
also what size is the element? looks like it's in a col-2 or col-3

**Comment by ldaug** [comment]
It is in a div that is 350px. But within the div is an accordion, and then each accordion item contains the select, so with the padding the select is about 287px.

    <div class="panel-body">
    <div class="form-group">
		<select class="form-control ng-pristine ng-untouched ng-valid ng-empty" size="6" style="overflow: auto; width:287px">
			<option style="display:none" value="" class="" selected="selected"></option>
			<option label="Abdulaziz al-Omari ⇒ Attended ⇒ FlightSafety International" value="object:274">Abdulaziz al-Omari ⇒ Attended ⇒ FlightSafety International</option>
			<option label="Abdulaziz al-Omari ⇒ Attended ⇒ Al Farouq Training Camp" value="object:275">Abdulaziz al-Omari ⇒ Attended ⇒ Al Farouq Training Camp</option>
			<option label="Abdulaziz al-Omari ⇒ Attended ⇒ Al Matar Training Camp" value="object:276">Abdulaziz al-Omari ⇒ Attended ⇒ Al Matar Training Camp</option>
			<option label="Abdulaziz al-Omari ⇒ Attended ⇒ Kandahar Meeting Summer 2000" value="object:277">Abdulaziz al-Omari ⇒ Attended ⇒ Kandahar Meeting Summer 2000</option>
			<option label="Ahmed al-Ghamdi ⇒ Attended ⇒ Al Farouq Training Camp" value="object:278">Ahmed al-Ghamdi ⇒ Attended ⇒ Al Farouq Training Camp</option>
		</select>
	</div>
</div>

**Comment by danielcollettepb** [comment]
Playing with the values in the Bootstrap site shows the same "selected" padding... will need to investigate if this can be changed:

![image](https://cloud.githubusercontent.com/assets/6116867/19326973/d2356fac-909a-11e6-8f5d-0f7c199a5bea.png)

**Comment by ldaug** [comment]
Just looked at this in IE and the scrollbar on x axis does not even appear. Had not noticed that before.

**Comment by danielcollettepb** [comment]
I'm not seeing it on chrome on mac either...

**Comment by ldaug** [comment]
Shows up in this example: https://plnkr.co/edit/NAyLHCH0l12Wl5fpI8if?p=preview

**Comment by ldaug** [comment]
A tooltip might mitigate the need for the scroll on x, but I did not get it working using ng-repeat. Maybe needs some jquery...

**Comment by danielcollettepb** [comment]
What about a simple title attr:

```<option ng-repeat="i in [1,2,3,4,5,6,7,8,9,10]" value="i" title="Lorem ipsum dolor sit amet {{i}}">Lorem ipsum dolor sit amet {{::i}}</option>```

**Comment by danielcollettepb** [comment]
and I got overflow working... operator error

**Comment by danielcollettepb** [comment]
looks like setting `width:auto` on the `<option>` fixes the selected padding:

![image](https://cloud.githubusercontent.com/assets/6116867/19327766/8ca73804-909e-11e6-8468-9cf11fb21d39.png)

**Comment by danielcollettepb** [comment]
Here's what I have so far, check out the `styles.scss` file:

http://plnkr.co/edit/4v5g6nrjU7NdZn1rfkTV?p=preview

**Comment by ldaug** [comment]
This does not work. If you make the string just a little longer, it starts clipping again.

![image](https://cloud.githubusercontent.com/assets/7534439/19350102/6bd25c18-9124-11e6-9b44-aa4b608e4a0c.png)

**Comment by danielcollettepb** [comment]
yes, you're right @ldaug... I discussed this with @SamSwartzPB and she is going to look into this as well. We're thinking that this _may_ need to be replaced with another control. Thinking about the tooltip (or title attr), even that would be clunky since the strings are so long.

Can you send us a screenshot via email with the full screen where this appears so we have some context?
